pkg/init-tools.sh

```diff
@@ -13,39 +13,84 @@ __BUILD_TOOLS_PATH=$__PACKAGES_DIR/Microsoft.DotNet.BuildTools/$__BUILD_TOOLS_PA
 __PROJECT_JSON_PATH=$__TOOLRUNTIME_DIR/$__BUILD_TOOLS_PACKAGE_VERSION
 __PROJECT_JSON_FILE=$__PROJECT_JSON_PATH/project.json
 __PROJECT_JSON_CONTENTS="{ \"dependencies\": { \"Microsoft.DotNet.BuildTools\": \"$__BUILD_TOOLS_PACKAGE_VERSION\" }, \"frameworks\": { \"netcoreapp1.0\": { } } }"
+__INIT_TOOLS_DONE_MARKER=$__PROJECT_JSON_PATH/done
 
-OSName=$(uname -s)
-case $OSName in
-    Darwin)
-        OS=OSX
-        __DOTNET_PKG=dotnet-dev-osx-x64
-        ulimit -n 2048
-        ;;
+# Extended version of platform detection logic from dotnet/cli/scripts/obtain/dotnet-install.sh 16692fc
+get_current_linux_name() {
```

Comment:
This way mirrors what we do in corefx because we don't have access to dotnet-install there. I think we want to remove dotnet-install from core-setup in the future, anyway, when it's changed to use buildtools.